polars/polars-core/src/chunked_array/ops/unique/rank.rs

```diff
@@ -328,30 +342,49 @@ mod test {
                 Some(8),
             ],
         );
-        let out = rank(&s, RankMethod::Max)
+        let out = rank(&s, RankMethod::Max, false)
             .u32()?
             .into_no_null_iter()
             .collect::<Vec<_>>();
-        assert_eq!(out, &[4, 5, 3, 8, 7, 3, 1, 6]);
+        assert_eq!(out, &[5, 6, 4, 1, 8, 4, 2, 7]);
```

Comment:
Why has this output changed?

Reply:
I changed null to rank highest, so that means null goes from rank 8 (lowest) to rank 1 (highest), which shifts all other ranks 1 lower (1->2, 2->3, etc.) It's like rotating the ranks. This _is_ a pretty gnarly change (could silently break some code that depends on nulls being ranked lowest), but it's probably better to do it sooner rather than later.